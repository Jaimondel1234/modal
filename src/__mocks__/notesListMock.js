export const notesListMock = [
  {
    id: 125465,
    title: 'first element',
    description: 'description first element',
    imgUrl: '',
  },
  {
    id: 125466,
    title: 'second element',
    description: 'description second element',
    imgUrl: '',
  },
  {
    id: 125467,
    title: 'third element',
    description: 'description third element',
    imgUrl: '',
  },
];
