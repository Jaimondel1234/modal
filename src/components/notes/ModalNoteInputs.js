import React from "react";
import Swal from "sweetalert2";
import { noteAddNew, noteUpdate, noteSetActive } from "../../actions/note";
import { useForm } from "../../hooks/useForm";
import styled from "styled-components";
import { InputStyle, FlexStyle, ButtonStyle } from "../../styles/core";
import { useDispatch, useSelector } from "react-redux";
import { uiCloseModal } from "../../actions/ui";

/** Style of modal input */
const ModalInputStyle = styled(InputStyle)`
  border-radius: 7px;
  font-size: 30px;
  margin-bottom: 20px;
  width: 95%;
  padding: 15px;
  padding-left: 25px;
`;

/*
 Modal Note Input
*/
export const ModalNoteInputs = () => {
  const dispatch = useDispatch();
  const { activeNote } = useSelector((state) => state.notes);
  const defaultEvent = {
    title: "",
    description: "",
    imgUrl: "",
  };

  const initEvent = activeNote || defaultEvent;
  const [formValues, handleInputChange, reset] = useForm(initEvent);

  const { description, title, imgUrl } = formValues;

  const closeModal = () => {
    dispatch(uiCloseModal());

    if (activeNote) {
      dispatch(noteUpdate(formValues));
      dispatch(noteSetActive(false));
    } else {
      const noteWithId = { id: new Date().getTime(), ...formValues };
      dispatch(noteAddNew([noteWithId]));
    }

    reset(initEvent);
  };
  const handleSubmitForm = (e) => {
    e.preventDefault();
    if (title.trim().length === 0) {
      Swal.fire("The title is required", "", "error");
    } else if (description.trim().length === 0) {
      Swal.fire("The description is required", "", "error");
    } else {
      Swal.fire("Note Saved", "", "success");
      closeModal();
    }
  };
  return (
    <FlexStyle
      as='form'
      alignItems='center'
      flexDirection='column'
      onSubmit={handleSubmitForm}
    >
      <ModalInputStyle
        type='text'
        placeholder='Title'
        name='title'
        autoComplete='off'
        value={title}
        onChange={handleInputChange}
        mt='30px'
      />

      <ModalInputStyle
        as='textarea'
        type='text'
        placeholder='Description'
        rows='5'
        name='description'
        value={description}
        onChange={handleInputChange}
        p='10px'
      />

      <ModalInputStyle
        type='text'
        placeholder='Imagen (Url)'
        name='imgUrl'
        autoComplete='off'
        value={imgUrl}
        onChange={handleInputChange}
      />

      <ButtonStyle
        p='15px'
        fontSize='30px'
        fontWeight='bold'
        letterSpacing='3px'
        lineHeight='40px'
        width='100%'
        type='submit'
      >
        Save
      </ButtonStyle>
    </FlexStyle>
  );
};
