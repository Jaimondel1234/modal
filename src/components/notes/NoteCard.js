import React, { useState } from "react";
import {
  PanelStyle,
  ImageStyle,
  TextStyle,
  ImgText,
  ImgContainer,
} from "../../styles/core";
import PropTypes from "prop-types";
import { DeleteEventFab } from "../ui/DeleteEventFab";
import { ModifyNoteFab } from "../ui/ModifyNoteFab";

/*
 Card with the information of a note
*/
export const NoteCard = ({ ...note }) => {
  const [isMouveIn, setIsMouseIn] = useState(false);

  const { id, title, imgUrl, description } = note;

  const handleMouseOver = () => {
    setIsMouseIn(true);
  };

  const handleMouseOut = () => {
    setIsMouseIn(false);
  };

  return (
    <PanelStyle
      ml='20px'
      mr='20px'
      width='450px'
      mb='50px'
      borderRadius='10px'
      onMouseOver={handleMouseOver}
      onMouseLeave={handleMouseOut}
    >
      <ImgContainer>
        <ImageStyle
          src={imgUrl}
          width='450px'
          height='400px'
          title={title}
          onError={(e) => {
            e.target.onerror = null;
            e.target.src =
              "https://www.escapadarural.com/blog/wp-content/uploads/2019/08/Playa-de-la-Tejita.-Tenerife.jpg";
          }}
        />
        {isMouveIn && <DeleteEventFab noteId={id} />}
        {isMouveIn && <ModifyNoteFab note={note} />}
        <ImgText color='white' fontSize='30px' fontWeight='bold'>
          {title}
        </ImgText>
      </ImgContainer>
      <TextStyle mb='30px' mr='25px' ml='30px' fontSize='25px'>
        {description}
      </TextStyle>
    </PanelStyle>
  );
};

NoteCard.propTypes = {
  note: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    imgUrl: PropTypes.string,
    description: PropTypes.string,
  }),
};
