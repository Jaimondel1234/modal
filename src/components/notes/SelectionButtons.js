import React from "react";
import styled from "styled-components";
import { FlexStyle, ButtonStyle } from "../../styles/core";
import { useDispatch } from "react-redux";
import { setListMode } from "../../actions/note";

const SelectorButtonStyle = styled(ButtonStyle)`
  padding: 7px;
  font-size: 30px;
  font-weight: bold;
  letter-spacing: 3px;
  margin-right: 10px;
`;

/**
 * It shows the buttons of selections (asc, desc, ..)
 */
export const SelectionButtons = () => {
  const dispatch = useDispatch();

  const handleClick = (listMode) => () => {
    dispatch(setListMode(listMode));
  };
  return (
    <FlexStyle alignItem='center' justifyContent='center' flexDirection='row'>
      <SelectorButtonStyle onClick={handleClick("oldToNew")}>
        ASC
      </SelectorButtonStyle>
      <SelectorButtonStyle onClick={handleClick("newToOld")}>
        DESC
      </SelectorButtonStyle>
      <SelectorButtonStyle onClick={handleClick("AToZ")}>
        A-Z
      </SelectorButtonStyle>
      <SelectorButtonStyle onClick={handleClick("ZToA")}>
        Z-A
      </SelectorButtonStyle>
    </FlexStyle>
  );
};
