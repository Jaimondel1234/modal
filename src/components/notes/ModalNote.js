import React from 'react';
import Modal from 'react-modal';
import { TextStyle } from '../../styles/core';
import { useSelector } from 'react-redux';
import { ModalNoteInputs } from './ModalNoteInputs';

const customStyles = {
  content: {
    top: '50%',
    left: '50%',
    right: 'auto',
    bottom: 'auto',
    marginRight: '-50%',
    transform: 'translate(-50%, -50%)',
    width: '50vw',
    padding: '50px',
  },
};

/**
 * Component of modal to create or update a note
 */
export const ModalNote = () => {
  if (process.env.NODE_ENV !== 'test') {
    Modal.setAppElement('#root');
  }
  const { modalOpen } = useSelector((state) => state.ui);
  const { activeNote } = useSelector((state) => state.notes);

  return (
    <Modal
      isOpen={modalOpen}
      style={customStyles}
      closeTimeoutMS={200}
      ariaHideApp={!(process.env.NODE_ENV === 'test')}
    >
      <TextStyle fontSize='60px' fontWeight='bold' textAlign='center'>
        {activeNote ? 'Change your Note' : 'Create your Note'}
      </TextStyle>
      <hr />
      <ModalNoteInputs />
    </Modal>
  );
};
