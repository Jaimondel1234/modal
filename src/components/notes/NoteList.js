import React from "react";
import { FlexStyle } from "../../styles/core";
import map from "lodash/map";
import { NoteCard } from "./NoteCard";
import { useSelector } from "react-redux";
import { getListByMode } from "../../helpers/helpers";

/*
 Cards List
*/
export const NoteList = () => {
  const { notes, listMode } = useSelector((state) => state.notes);
  let noteListToShow = getListByMode(listMode, notes);

  return (
    <FlexStyle
      alignItem='center'
      justifyContent='center'
      flexDirection='column'
      mt='30px'
    >
      <FlexStyle
        alignItem='center'
        justifyContent='center'
        flexDirection='row'
        flexWrap='wrap'
      >
        {noteListToShow &&
          noteListToShow.length > 0 &&
          map(noteListToShow, (props) => (
            <NoteCard key={props.id} {...props} />
          ))}
      </FlexStyle>
    </FlexStyle>
  );
};
