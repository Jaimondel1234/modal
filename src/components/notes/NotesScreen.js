import React, { useEffect } from "react";
import { NoteList } from "./NoteList";
import { ContainerStyle } from "../../styles/core";
import { AddNewFab } from "../ui/AddNewFab";
import { ModalNote } from "./ModalNote";
import { useSelector } from "react-redux";
import { ButtonsSelector } from "./ButtonsSelector";

/*
  Component of Screen Notes
*/
export const NotesScreen = () => {
  const { notes } = useSelector((state) => state.notes);

  useEffect(() => {
    localStorage.setItem("notes", JSON.stringify(notes));
  }, [notes]);

  const { modalOpen } = useSelector((state) => state.ui);
  return (
    <>
      <ContainerStyle mt='20px'>
        <ButtonsSelector />
        <NoteList />
        <AddNewFab />
      </ContainerStyle>
      {modalOpen && <ModalNote />}
    </>
  );
};
