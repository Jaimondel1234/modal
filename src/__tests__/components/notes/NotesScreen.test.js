import React from "react";
import { NotesScreen } from "../../../components/notes/NotesScreen";
import { notesListMock } from "../../../__mocks__/notesListMock";
import configureStore from "redux-mock-store";
import { mount } from "enzyme";
import { Provider } from "react-redux";

describe("Test <NotesScreen/>", () => {
  const mockStore = configureStore([]);
  const initState = {
    notes: {
      notes: notesListMock,
    },
    ui: {
      modalOpen: true,
    },
  };

  const store = mockStore(initState);

  const wrapper = mount(
    <Provider store={store}>
      <NotesScreen />
    </Provider>
  );
  test("it must be displayed correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("if modal open true, the modal must be showed", () => {
    expect(wrapper.find("ModalNote").at(0).exists()).toBe(true);
  });
});
