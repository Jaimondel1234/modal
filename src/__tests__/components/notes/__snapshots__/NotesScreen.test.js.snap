// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <NotesScreen/> it must be displayed correctly 1`] = `
Array [
  <div
    className="core__ContainerStyle-sc-1mkz20q-0 eIDSUq"
  >
    <div
      className="core__FlexStyle-sc-1mkz20q-3 hkFmaH"
    >
      <button
        className="core__ButtonStyle-sc-1mkz20q-5 SelectionButtons__SelectorButtonStyle-sc-11fith9-0 QFcyH kzXbYk"
        onClick={[Function]}
      >
        ASC
      </button>
      <button
        className="core__ButtonStyle-sc-1mkz20q-5 SelectionButtons__SelectorButtonStyle-sc-11fith9-0 QFcyH kzXbYk"
        onClick={[Function]}
      >
        DESC
      </button>
      <button
        className="core__ButtonStyle-sc-1mkz20q-5 SelectionButtons__SelectorButtonStyle-sc-11fith9-0 QFcyH kzXbYk"
        onClick={[Function]}
      >
        A-Z
      </button>
      <button
        className="core__ButtonStyle-sc-1mkz20q-5 SelectionButtons__SelectorButtonStyle-sc-11fith9-0 QFcyH kzXbYk"
        onClick={[Function]}
      >
        Z-A
      </button>
    </div>
    <div
      className="core__FlexStyle-sc-1mkz20q-3 fhvbtd"
    >
      <div
        className="core__FlexStyle-sc-1mkz20q-3 gvOifw"
      >
        <div
          className="core__PanelStyle-sc-1mkz20q-1 bRoYup"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          width="450px"
        >
          <div
            className="core__ImgContainer-sc-1mkz20q-10 iknRaK"
          >
            <img
              className="core__ImageStyle-sc-1mkz20q-6 kvKmGh"
              height="400px"
              onError={[Function]}
              src=""
              title="first element"
              width="450px"
            />
            <div
              className="core__TextStyle-sc-1mkz20q-2 core__ImgText-sc-1mkz20q-11 gGsHdO fgsaRG"
              color="white"
              fontSize="30px"
              fontWeight="bold"
            >
              first element
            </div>
          </div>
          <div
            className="core__TextStyle-sc-1mkz20q-2 lpdOGy"
            fontSize="25px"
          >
            description first element
          </div>
        </div>
        <div
          className="core__PanelStyle-sc-1mkz20q-1 bRoYup"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          width="450px"
        >
          <div
            className="core__ImgContainer-sc-1mkz20q-10 iknRaK"
          >
            <img
              className="core__ImageStyle-sc-1mkz20q-6 kvKmGh"
              height="400px"
              onError={[Function]}
              src=""
              title="second element"
              width="450px"
            />
            <div
              className="core__TextStyle-sc-1mkz20q-2 core__ImgText-sc-1mkz20q-11 gGsHdO fgsaRG"
              color="white"
              fontSize="30px"
              fontWeight="bold"
            >
              second element
            </div>
          </div>
          <div
            className="core__TextStyle-sc-1mkz20q-2 lpdOGy"
            fontSize="25px"
          >
            description second element
          </div>
        </div>
        <div
          className="core__PanelStyle-sc-1mkz20q-1 bRoYup"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          width="450px"
        >
          <div
            className="core__ImgContainer-sc-1mkz20q-10 iknRaK"
          >
            <img
              className="core__ImageStyle-sc-1mkz20q-6 kvKmGh"
              height="400px"
              onError={[Function]}
              src=""
              title="third element"
              width="450px"
            />
            <div
              className="core__TextStyle-sc-1mkz20q-2 core__ImgText-sc-1mkz20q-11 gGsHdO fgsaRG"
              color="white"
              fontSize="30px"
              fontWeight="bold"
            >
              third element
            </div>
          </div>
          <div
            className="core__TextStyle-sc-1mkz20q-2 lpdOGy"
            fontSize="25px"
          >
            description third element
          </div>
        </div>
      </div>
    </div>
    <button
      className="core__FabStyle-sc-1mkz20q-9 AddNewFab__PlusFabStyle-sc-1cvbv4n-0 irwEJW dRoibW"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus fa-w-14 "
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </div>,
  <Portal
    containerInfo={
      <div
        class="ReactModalPortal"
      >
        <div
          class="ReactModal__Overlay ReactModal__Overlay--after-open"
          style="position: fixed; top: 0px; left: 0px; right: 0px; bottom: 0px; background-color: rgba(255, 255, 255, 0.75);"
        >
          <div
            class="ReactModal__Content ReactModal__Content--after-open"
            role="dialog"
            style="position: absolute; top: 50%; left: 50%; border: 1px solid #ccc; background: rgb(255, 255, 255); overflow: auto; border-radius: 4px; outline: none; padding: 50px; margin-right: -50%; transform: translate(-50%, -50%); width: 50vw;"
            tabindex="-1"
          >
            <div
              class="core__TextStyle-sc-1mkz20q-2 kA-dAxr"
              font-size="60px"
              font-weight="bold"
            >
              Create your Note
            </div>
            <hr />
            <form
              class="core__FlexStyle-sc-1mkz20q-3 iTHhnD"
            >
              <input
                autocomplete="off"
                class="core__InputStyle-sc-1mkz20q-4 ModalNoteInputs__ModalInputStyle-sc-18t1odo-0 bMnAXg ksglXK"
                name="title"
                placeholder="Title"
                type="text"
                value=""
              />
              <textarea
                class="core__InputStyle-sc-1mkz20q-4 ModalNoteInputs__ModalInputStyle-sc-18t1odo-0 fsMNMP ksglXK"
                name="description"
                placeholder="Description"
                rows="5"
                type="text"
              />
              <input
                autocomplete="off"
                class="core__InputStyle-sc-1mkz20q-4 ModalNoteInputs__ModalInputStyle-sc-18t1odo-0 kivKiO ksglXK"
                name="imgUrl"
                placeholder="Imagen (Url)"
                type="text"
                value=""
              />
              <button
                class="core__ButtonStyle-sc-1mkz20q-5 cgFcMK"
                font-size="30px"
                font-weight="bold"
                letter-spacing="3px"
                type="submit"
                width="100%"
              >
                Save
              </button>
            </form>
          </div>
        </div>
      </div>
    }
  >
    <div
      className="ReactModal__Overlay ReactModal__Overlay--after-open"
      onClick={[Function]}
      onMouseDown={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.75)",
          "bottom": 0,
          "left": 0,
          "position": "fixed",
          "right": 0,
          "top": 0,
        }
      }
    >
      <div
        className="ReactModal__Content ReactModal__Content--after-open"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        role="dialog"
        style={
          Object {
            "WebkitOverflowScrolling": "touch",
            "background": "#fff",
            "border": "1px solid #ccc",
            "borderRadius": "4px",
            "bottom": "auto",
            "left": "50%",
            "marginRight": "-50%",
            "outline": "none",
            "overflow": "auto",
            "padding": "50px",
            "position": "absolute",
            "right": "auto",
            "top": "50%",
            "transform": "translate(-50%, -50%)",
            "width": "50vw",
          }
        }
        tabIndex="-1"
      >
        <div
          className="core__TextStyle-sc-1mkz20q-2 kA-dAxr"
          fontSize="60px"
          fontWeight="bold"
        >
          Create your Note
        </div>
        <hr />
        <form
          className="core__FlexStyle-sc-1mkz20q-3 iTHhnD"
          onSubmit={[Function]}
        >
          <input
            autoComplete="off"
            className="core__InputStyle-sc-1mkz20q-4 ModalNoteInputs__ModalInputStyle-sc-18t1odo-0 bMnAXg ksglXK"
            name="title"
            onChange={[Function]}
            placeholder="Title"
            type="text"
            value=""
          />
          <textarea
            className="core__InputStyle-sc-1mkz20q-4 ModalNoteInputs__ModalInputStyle-sc-18t1odo-0 fsMNMP ksglXK"
            name="description"
            onChange={[Function]}
            placeholder="Description"
            rows="5"
            type="text"
            value=""
          />
          <input
            autoComplete="off"
            className="core__InputStyle-sc-1mkz20q-4 ModalNoteInputs__ModalInputStyle-sc-18t1odo-0 kivKiO ksglXK"
            name="imgUrl"
            onChange={[Function]}
            placeholder="Imagen (Url)"
            type="text"
            value=""
          />
          <button
            className="core__ButtonStyle-sc-1mkz20q-5 cgFcMK"
            fontSize="30px"
            fontWeight="bold"
            letterSpacing="3px"
            type="submit"
            width="100%"
          >
            Save
          </button>
        </form>
      </div>
    </div>
  </Portal>,
]
`;
